File: src/types/connection.ts
export type SslMode = 'disable' | 'require' | 'verify-ca' | 'verify-full';
export type SshAuthMethod = 'password' | 'private-key' | 'agent';

export interface SshConfig {
  host: string;
  port: number;
  username: string;
  auth_method: SshAuthMethod;
  password?: string;
  private_key_path?: string;
  passphrase?: string;
}

export interface ConnectionConfig {
  id: string;
  name: string;
  color_tag?: string;
  group?: string;
  host: string;
  port: number;
  database: string;
  username: string;
  password?: string;
  save_password?: boolean;
  ssl_mode: SslMode;
  client_cert_path?: string;
  client_key_path?: string;
  ca_cert_path?: string;
  ssh_enabled: boolean;
  ssh_config?: SshConfig;
}

File: src-tauri/src/models/connection.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone, Copy, PartialEq, Eq)]
#[serde(rename_all = "kebab-case")]
pub enum SslMode {
    Disable,
    Require,
    VerifyCa,
    VerifyFull,
}

#[derive(Debug, Serialize, Deserialize, Clone, Copy, PartialEq, Eq)]
#[serde(rename_all = "kebab-case")]
pub enum SshAuthMethod {
    Password,
    PrivateKey,
    Agent,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct SshConfig {
    pub host: String,
    pub port: u16,
    pub username: String,
    pub auth_method: SshAuthMethod,
    pub password: Option<String>,
    pub private_key_path: Option<String>,
    pub passphrase: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ConnectionConfig {
    pub id: String,
    pub name: String,
    pub color_tag: Option<String>,
    pub group: Option<String>,
    pub host: String,
    pub port: u16,
    pub database: String,
    pub username: String,
    pub password: Option<String>,
    #[serde(default = "default_save_password")]
    pub save_password: bool,
    pub ssl_mode: SslMode,
    pub client_cert_path: Option<String>,
    pub client_key_path: Option<String>,
    pub ca_cert_path: Option<String>,
    pub ssh_enabled: bool,
    pub ssh_config: Option<SshConfig>,
}

fn default_save_password() -> bool {
    true
}
